### PR TITLE
Create CallingCard.lua

### PR DESCRIPTION
**Pull request by BBBalatroMod**
Contains the code for the Calling Card joker

### CODE REVIEW

**Comment by GitNether** [comment]
The addition to the config in the main `paperback.lua` file is missing (it won't be initialized otherwise)

**Comment by OppositeWolf770** [comment]
I was having issues getting this joker to load in its current state. I enabled it in the config but it kept giving me an error saying that the `end_calculate_context` field was null. I did some research and asked on the Discord and they said to change it to `if context.joker_main` and add `card` in as a function parameter to calculate. Here is the working code if you want to take a look:

```
SMODS.Atlas {  
    key = 'callingcardimg',
    px = 71,
    py = 95,
    path = 'CallingCard.png'
}

SMODS.Joker {
    key = 'callingcard',
    loc_txt = {
        name = "Calling Card",                                --name used by the joker
        text = {
            "Whenever you defeat a {C:attention}Boss Blind{} or activate its {C:attention}effect{}, this Joker gains {X:red,C:white}0.25X{}",             --description text.		
            "{C:inactive}(currently gives {}{X:red,C:white}#2#X{}{C:inactive} Mult){}"                       --more than 5 lines look odd
        }
    },
    config = { extra = { mult = 0, x_mult = 1 } }, --variables used for abilities and effects.
    rarity = 2,                                         --rarity 1=common, 2=uncommen, 3=rare, 4=legendary
    pos = { x = 0, y = 0 },                             --pos in spirtesheet 0,0 for single sprites or the first sprite in the spritesheet
    atlas = 'callingcardimg',                                        --defines the atlas that you want to use for the sprite sheet. atlas=nil if you want to use single sprites
    cost = 6,                                           --cost to buy the joker in shops
    unlocked = true,                                    --joker is unlocked by default
    discovered = true,                                  --joker is discovered by default
    blueprint_compat = true,                            --does joker work with blueprint
    eternal_compat = true,                              --can joker be eternal
    soul_pos = nil,                            --pos of a soul sprite.

    calculate = function(self, card, context)                 --define calculate functions here
        if G.GAME.blind.triggered and not context.blueprint then 
            card.ability.extra.x_mult = card.ability.extra.x_mult + 0.25
        end
        if context.end_of_round and not context.individual and not context.repetition and not context.blueprint and G.GAME.blind.boss and not self.gone then
            card.ability.extra.x_mult = card.ability.extra.x_mult + 0.25
        end
        if context.joker_main then
            return {
                Xmult_mod = card.ability.extra.x_mult,
                card = card,
                message = localize { type = 'variable', key = 'a_xmult', vars = { card.ability.extra.x_mult } }
            }
        end
    end,

    loc_vars = function(self, info_queue, center) --defines variables to use in the UI. you can use #1# for example to show the mult variable, and #2# for x_mult
        return { vars = {center.ability.extra.mult, center.ability.extra.x_mult} }
    end,
}
```

File: joker/CallingCard.lua [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    eternal_compat = true,                              --can joker be eternal
+    soul_pos = nil,                            --pos of a soul sprite.
+
+    calculate = function(self, context)                 --define calculate functions here
+        if G.GAME.blind.triggered and not context.blueprint then 
+            card.ability.extra.x_mult = card.ability.extra.x_mult + 0.25
+        end
+        if context.end_of_round and not context.individual and not context.repetition and not context.blueprint and G.GAME.blind.boss and not self.gone then
+            card.ability.extra.x_mult = card.ability.extra.x_mult + 0.25
+        end
```
Comment (OppositeWolf770) [inline joker/CallingCard.lua:34]: Something wrong was going on with these lines. When a boss blind was triggered, it was triggering this effect multiple times. It even happened about 5 times on the end_of_round part for me once. I'm not sure what the cause of that is.

File: joker/CallingCard.lua [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if SMODS.end_calculate_context(context) then
```
Comment (OppositeWolf770) [inline joker/CallingCard.lua:35]: This needs to be changed from `SMODS.end_calculate_context(context)` to `context.joker_main`

File: joker/CallingCard.lua [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    pos = { x = 0, y = 0 },                             --pos in spirtesheet 0,0 for single sprites or the first sprite in the spritesheet
+    atlas = 'callingcardimg',                                        --defines the atlas that you want to use for the sprite sheet. atlas=nil if you want to use single sprites
+    cost = 6,                                           --cost to buy the joker in shops
+    unlocked = true,                                    --joker is unlocked by default
+    discovered = true,                                  --joker is discovered by default
+    blueprint_compat = true,                            --does joker work with blueprint
+    eternal_compat = true,                              --can joker be eternal
+    soul_pos = nil,                            --pos of a soul sprite.
+
+    calculate = function(self, context)                 --define calculate functions here
```
Comment (OppositeWolf770) [inline joker/CallingCard.lua:28]: This needs to include `card` as a parameter between `self` and `context`

File: joker/CallingCard.lua (hunk 2 above)
Comment (OppositeWolf770) [reply joker/CallingCard.lua:35]: This was according to someone on Discord, when I was having issues getting the Joker to work.

**Comment by GitNether** [comment]
Also, pull main and merge, since we removed the individual sprites and replaced it with one sprite sheet

**Comment by OppositeWolf770** [comment]
Also consider changing the filename to calling_card.lua, since basically everything else is snake_case.

File: joker/CallingCard.lua (hunk 1 above)
Comment (GitNether) [reply joker/CallingCard.lua:34]: `context.joker_main` was missing. after alot of debugging we solved it together earlier

**Comment by GitNether** [comment]
Moving to branch [add-calling-card](https://github.com/GitNether/paperback/tree/add-calling-card) to merge with refactoring

**Pull request closed without merging**